services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/extensions/VersionedRecordExtensionTest.java

```diff
@@ -163,6 +163,25 @@ public void beforeWrite_returnsNoOpModification_ifVersionAttributeNotDefined() {
         assertThat(writeModification, is(WriteModification.builder().build()));
     }
 
+    @Test
+    public void beforeWrite_returnsNoOpModification_ifOperationIsBatchWrite() {
+        FakeItem fakeItem = createUniqueFakeItem();
+        fakeItem.setVersion(13);
+        Map<String, AttributeValue> fakeItemWithInitialVersion =
+            new HashMap<>(FakeItem.getTableSchema().itemToMap(fakeItem, true));
+        fakeItemWithInitialVersion.put("version", AttributeValue.builder().n("14").build());
+
+        WriteModification result =
+            versionedRecordExtension.beforeWrite(DefaultDynamoDbExtensionContext
+                                                     .builder()
+                                                     .items(FakeItem.getTableSchema().itemToMap(fakeItem, true))
+                                                     .tableMetadata(FakeItem.getTableMetadata())
+                                                     .operationName(OperationName.BATCH_WRITE_ITEM)
```

Comment:
The SDK sends the individual operation name, e.g `PUT_ITEM`, making this never work 😟